tough/src/key_source.rs

```diff
@@ -17,11 +17,38 @@ pub trait KeySource: Debug + Send + Sync {
     fn as_sign(&self) -> Result<Box<dyn Sign>, Box<dyn std::error::Error + Send + Sync + 'static>>;
 
     /// Writes a key back to the `KeySource`
-    fn write(
+    fn write(&self, value: &str) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>>;
```

Comment:
I think the only reason we have a `write` method on this trait is to account for `tuftool`'s usage of it for SSM, right?  Perhaps it makes sense to evaluate what it means to eliminate this method from the trait altogether.  

@iliana : thoughts?

Reply:
write is used by SSM to store the rsa key in a parameter.
write is used by KMS to assign alias to the generated Key.
write is used by localsource to write generated key in a file.


Previously, gen-rsa-key command creates the rsa-key pair and calls write method.
Now, gen-rsa-key calls the create method of keysource, which in turn calls the write method. After this code change, write method is not longer called directly. We can get rid of it or make it private.  

gen-rsa-key command is for convenience, but it helps in end to end testing and programmatically create key.

I was also wondering does it make sense to call 'add-key' command from 'gen-rsa-key' ? We have separate command 'add-key', which user can use if they wan't to add the key. Or we can all command 'gen-add-rsa-key'. :P

Reply:
> gen-rsa-key command is for convenience, but it helps in end to end testing and programmatically create key.

I think we should take a closer look at this design, there may be another way. Generally speaking we shouldn't be adding too many dependencies or assumptions to the core library (`tough`).